gcovr/gcov.py

```diff
@@ -25,6 +25,44 @@
 c_style_comment_pattern = re.compile('/\*.*?\*/')
 cpp_style_comment_pattern = re.compile('//.*?$')
 
+gcov_required_options = [
+    "--branch-counts",
+    "--branch-probabilities",
+    "--preserve-paths",
+    '--object-directory']
+
+
+class GCovSupportChecker(object):
+    """Check that gcov supports the options we need"""
+    def __init__(self):
+        self.checked = {}
+
+    def check(self, gcov_path):
+        if gcov_path not in self.checked:
+            cmd = gcov_path.split()
+            cmd.append("--help")
+            self.checked[gcov_path] = False
+            try:
+                help = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+            except subprocess.CalledProcessError as cpe:
+                help = cpe.output
+            help = str(help)
+            for option in gcov_required_options:
+                if option not in help:
+                    try:
+                        gcov_path = subprocess.check_output(
+                            ["which", gcov_path]).strip().decode("utf-8")
```

Comment:
I think this will fail if `gcov_path` contains any extra options, as may be necessary when using gcovr with Clang (`--gcov-executable='llvm-cov gcov'`). Instead perhaps `gcov_path.split()[0]`.

Reply:
Yes, maybe the `which` test to print out what the program really was is overkill.